mapper/dm_agent/src/time_provider.rs

```diff
@@ -0,0 +1,17 @@
+use chrono::{DateTime, FixedOffset, Local};
+
+pub type Timestamp = DateTime<FixedOffset>;
+
+pub trait TimeProvider: Sync + Send + 'static {
+    fn now(&self) -> Timestamp;
+}
+
+#[derive(Clone)]
+pub struct SystemTimeProvider;
+
+impl TimeProvider for SystemTimeProvider {
+    fn now(&self) -> Timestamp {
+        let local_time_now = Local::now();
+        local_time_now.with_timezone(local_time_now.offset())
+    }
+}
```

Comment:
This is a good addition. So useful that it deserves to be moved in `common`.

However:
* I would rather name the trait `Clock` and the implementation `WallClock` to stress this is the local time.
* If moved in `common`, one must also removed the `current_timestamp()` function from `thin_edge_json/src/measurement.rs`.